src/containers/AccountPin.js

```diff
@@ -6,17 +6,17 @@ import { Actions } from 'react-native-router-flux'
 import AccountPin from '../components/AccountPin'
 import { addAccount, setPin } from '../actions/accounts'
 import { signedTx } from '../actions/transactions'
-import { keccak, brainWalletSign } from '../util/native'
+import { brainWalletSign } from '../util/native'
 import { saveAccount } from '../util/db'
+import store from '../util/store'
 
 const mapStateToPropsEnterPin = (state, ownProps) => ({
-  account: state.accounts.selected,
-  extra: state.transactions.pendingTransaction.rlp
+  account: state.accounts.selected
 })
 
-async function signTransaction (dispatch, account, rlp) {
+async function signTransaction (dispatch, account) {
   try {
-    let hash = await keccak(rlp)
+    let hash = store.getState().transactions.pendingTransaction.rlpHash
```

Comment:
`redux-thunk` allows you to produce actions that are actually functions that look like this: `action = (dispatch, getState) => {}`. You can then create a chain of interconnected actions handling async calls correctly.

Under the hood it's just a simple middleware that literally just does this:
```js
state => next => action => {
  if (typeof action === 'function') {
    return action(dispatch, state.getState)
  }
  
  return next(action);
}
```
https://github.com/gaearon/redux-thunk/blob/master/src/index.js#L2

So you wouldn't need to import the whole `store` singleton and I think it would be a bit cleaner thanks to that (action could be re-used for other stores and wouldn't have direct dependency on it).